specs/Standards/NonFungibleToken/Event.md

```diff
@@ -0,0 +1,200 @@
+
+## Motivation
+
+NFT-driven apps perform many similar actions.
+For example - `minting`, `burning` and `transferring`.
+Each app may have their own way of performing these actions.
+This introduces inconsistency in capturing these events.
+This extension addresses that.
+
+It's common for NFT apps to transfer many or one token at a time.
+Other applications need to track these and similar events consistently.
+If not, tracking state across many NFT-driven apps become infeasible.
+
+We need a standard way to capture events.
+This was discussed here https://github.com/near/NEPs/issues/254.
+
+## Events
+
+Many apps use different interfaces that represent the same action.
+This interface standardizes that process by introducing event logs.
+There is no Event NEP yet, so this standard paves the road to that.
+
+Events use standard logs capability of NEAR and defined as a convention.
+Events are log entries that start with `EVENT_JSON:` prefix followed by a single valid JSON document of the following interface:
+
+```ts
+// Interface to capture data 
+// about an event
+// Arguments
+// * `standard`: name of standard e.g. nep171
+// * `version`: e.g. 1.0.0
+// * `event`: string
+// * `data`: associate event data
+interface EventLogData {
+    standard: string,
+    version: string,
+    event: string,
+    data?: unknown,
+}
+```
+
+#### Valid event logs:
+
+```js
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered"}
+```
+
+```js
+EVENT_JSON:{
+  "standard": "nepXXX",
+  "version": "1.0.0",
+  "event": "xyz_is_triggered"
+}
+```
+
+```js
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered", "data": {"triggered_by": "foundation.near"}}
+```
+
+#### Invalid event logs:
+
+* Two events in a single log entry (instead, call `log` for each individual event)
+```
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered"}
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered"}
+```
+* Invalid JSON data
+```
+EVENT_JSON:invalid json
+```
+
+## Interface
+
+Non-Fungible Token Events MUST have `standard` set to `"nep171"`, standard version set to `"1.0.0"`, `event` value is one of `nft_mint`, `nft_burn`, `nft_transfer`, and `data` must be of one of the following relavant types: `NftMintLog[] | NftTransferLog[] | NftBurnLog[]`:
+
+```ts
+interface EventLogData {
+    standard: "nep171",
+    version: "1.0.0",
+    event: "nft_mint" | "nft_burn" | "nft_transfer",
+    data: NftMintLog[] | NftTransferLog[] | NftBurnLog[],
+}
+```
+
+```ts
+// An event log to capture token minting
+// Arguments
+// * `owner_id`: "account.near"
+// * `token_ids`: ["1", "abc"]
+// * `memo`: optional message
+interface NftMintLog {
+    owner_id: string,
+    token_ids: string[],
+    memo?: string
+}
+
+// An event log to capture token burning
+// Arguments
+// * `owner_id`: owner of tokens to burn
+// * `authorised_id`: approved account to burn, if applicable
```

Comment:
```suggestion
// * `authorized_id`: approved account to burn, if applicable
```
Hate to be nit-picky about this, but I do see other references in the NEP to the word "authorize" and "authorized" and think this will be an introduction of a word spelling that is inconsistent with numerous cases in our docs, NEAR CLI, nearcore, Rust SDK, and other NEPs.

So I think we'll have to stick with the spelling we have in these other places for consistently, lest we cause confusion for end users.


<img width="1764" alt="Screen Shot 2021-10-21 at 2 25 37 PM" src="https://user-images.githubusercontent.com/1042667/138359406-82b1de93-589e-4897-9e7b-c820e38ce890.png">
<img width="944" alt="Screen Shot 2021-10-21 at 2 23 23 PM" src="https://user-images.githubusercontent.com/1042667/138359424-3a6ddf20-a4fa-4694-b7ac-adaa1c9b3385.png">
<img width="1359" alt="Screen Shot 2021-10-21 at 2 22 05 PM" src="https://user-images.githubusercontent.com/1042667/138359426-557f2c27-206a-483f-a521-03a62918a226.png">
![Screen Shot 2021-10-21 at 2 20 42 PM](https://user-images.githubusercontent.com/1042667/138359429-a5ea33e1-474a-4198-8546-0478dd07ef5b.png)
<img width="1322" alt="Screen Shot 2021-10-21 at 2 19 42 PM" src="https://user-images.githubusercontent.com/1042667/138359430-c540c6f0-9b01-41c7-b9f7-ef54d4d7d92c.png">